onnxruntime/test/providers/cpu/tensor/tensor_op_test.cc

```diff
@@ -277,6 +277,28 @@ TEST(TensorOpTest, CastFromFloat16) {
   TestCastOp(input, int64_t_data, shape, TensorProto::INT64);
 }
 
+TEST(TensorOpTest, CastFromString) {
+  const std::vector<int64_t> shape{2, 2, 2};
+  std::initializer_list<std::string> string_data = {"-inf", "+INF", "2.0f", "3.0f", "4.0f", "5.0f", "NaN", "nan"};
+  const std::initializer_list<float> float_output = {-(std::numeric_limits<float>::infinity()), std::numeric_limits<float>::infinity(), 2.0f, 3.0f, 4.0f, 5.0f, NAN, NAN};
+  TestCastOp(string_data, float_output, shape, TensorProto::FLOAT);
+
+  std::initializer_list<std::string> int_string_data = {"0", "1", "2", "3", "4", "5", "6", "7"};
```

Comment:
>7 [](start = 92, length = 1)

Tests need to cover edge cases, for example, max/min value of int64.

Reply:
Will do it after rewrite the kernel.

Reply:
Could you please test the conversion int64max/min-->string-->int64max/min and see if the two int64's are the same? If we have such a test, we could realize string-->double-->dst is not good earlier.

---
In reply to: [251635052](https://github.com/Microsoft/onnxruntime/pull/379#discussion_r251635052) [](ancestors = 251635052)